From a18a654148649023ee99abc21875f1f26f42b765 Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Mon, 9 Sep 2019 09:44:18 +0300
Subject: [PATCH 09/10] sb-imx8x: enable RS485 UART

RS485 half-duplex is implemented in the SB-iMX8X baseboard.
Enable the relevant UART.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/sb-imx8x.dts | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/compulab/sb-imx8x.dts b/arch/arm64/boot/dts/compulab/sb-imx8x.dts
index 9abb25afdcf7..a26454e0ba03 100644
--- a/arch/arm64/boot/dts/compulab/sb-imx8x.dts
+++ b/arch/arm64/boot/dts/compulab/sb-imx8x.dts
@@ -279,6 +279,13 @@
 	status = "okay";
 };
 
+/* RS485 HD */
+&lpuart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpuart2>;
+	status = "okay";
+};
+
 &iomuxc {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_hog_sb>;
@@ -374,5 +381,12 @@
 				SC_P_FLEXCAN2_RX_ADMA_FLEXCAN2_RX		0x00000021
 			>;
 		};
+
+		pinctrl_lpuart2: lpuart2grp {
+			fsl,pins = <
+				SC_P_UART2_RX_ADMA_UART2_RX			0x06000020
+				SC_P_UART2_TX_ADMA_UART2_TX			0x06000020
+			>;
+		};
 	};
 };
-- 
2.11.0

